rtfs.go

```diff
@@ -16,22 +16,32 @@ import (
 type IpfsManager struct {
 	shell       *ipfsapi.Shell
 	nodeAPIAddr string
+	token       string
+	direct      bool
 }
 
 // NewManager is used to initialize our Ipfs manager struct
-func NewManager(ipfsURL string, timeout time.Duration, direct bool) (*IpfsManager, error) {
+func NewManager(ipfsURL, token string, timeout time.Duration, direct bool) (*IpfsManager, error) {
```

Comment:
```suggestion
func NewManager(ipfsURL string, timeout time.Duration, direct bool) (*IpfsManager, error) {
```
instead:

```go
why not just remove the additional parameter from constructor, since it's already really bloated, and add:

```
func (im *IpfsManager) WithAuthorization(token string) {
    return &IpfsManager{
        shell: im.shell.WithAuthorization(token),
        // ... etc
    }
}
```

Reply:
authoriziation is, at most, a *per-request* thing - we should do the same thing I did with the shell, that is attach a constructor for generating authenticated clients.

Reply:
My one issue with this is if instantiating rtfs similar to the way we do with public networks in that a single instance is re-usable across multiple requests, say for integrating with a particular private network, needing to include the jwt each time.

Doing this allows for re-use across multiple different calls that may need to occur with a single instance without having to append `WithAuthorization` to every one.

Reply:
what we're doing here is essentially `WithAuthorization` - I think it's much safer, to avoid polluting our main client, and to make sure each authenticated call has the correct and up-to-date token, and to avoid the boilerplate currently present

Reply:
Thats a pretty good point. I'll make the change